pkg/controller/nodenetworkconfigurationpolicy/conditions/manager.go

```diff
@@ -49,19 +51,22 @@ func (m *Manager) NotifySuccess() {
 }
 
 func (m *Manager) updateEnactmentCondition(
-	conditionsSetter func(*nmstatev1alpha1.EnactmentList, string, string),
+	conditionsSetter func(*nmstatev1alpha1.NodeNetworkConfigurationEnactment, string),
 	message string,
 ) error {
 	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
-		instance := &nmstatev1alpha1.NodeNetworkConfigurationPolicy{}
-		err := m.client.Get(context.TODO(), m.policyName, instance)
+		instance := &nmstatev1alpha1.NodeNetworkConfigurationEnactment{}
+		err := m.client.Get(context.TODO(), nmstatev1alpha1.EnactmentKey(m.nodeName, m.policy.Name), instance)
 		if err != nil {
-			return err
+			return errors.Wrap(err, "getting enactment failed")
 		}
 
-		conditionsSetter(&instance.Status.Enactments, m.nodeName, message)
+		conditionsSetter(instance, message)
 
 		err = m.client.Status().Update(context.TODO(), instance)
-		return err
+		if err != nil {
+			return errors.Wrap(err, "updating enactment state failed")
```

Comment:
If you wrap the error, how does the `RetryOnConfict` knows that the callback failed on conflict?

Reply:
Depends on this:
```
// ReasonForError returns the HTTP status for a particular error.               
func ReasonForError(err error) metav1.StatusReason {                            
    switch t := err.(type) {                                                    
    case APIStatus:                                                             
        return t.Status().Reason                                                
    }                                                                           
    return metav1.StatusReasonUnknown                                           
}   
```

Reply:
So by wrapping you lose the granularity, don't you?